### ISSUE TITLE
Feature Request: ability to maintain multiple managers within one vault

### ISSUE THREAD

**Issue by roleforgaming** [carlonicora/obsidian-rpg-manager#5]
Would love for there to be an ability to manage multiple campaigns within one vault. As far as I see it, so far you can only manage one campaign overall, as you cannot assign unique tags that only call the notes that belong to each campaign.

**Comment by carlonicora** [comment]
Hi @roleforgaming 

I am working on the feature right now. The challenge I had was that every `outline` (Adventures, Sessions and Scenes) and every `element` (Player Characters, Non Player Characters, Factions, Locations, Events and Clues) must identify their campaign.

So far the most elegant way has been to require two tags. The first to identify `outline` and `elements` and the second one to identify the parent campaign.

The outline tags require the creation of an `id` and the specification of the id of the parent.

- Campaigns: `rpgm/outline/campaign/{campaignId}
- Adventures: `rpgm/outline/adventure/{campaignId}/{adventureId}`
- Sessions: `rpgm/outline/session/{adventureId}/{sessionId}`
- Scenes: `rpgm/outline/scene/{sessionId}/{sceneId}`

The elements require a specific tag to identify their types (but don't require an `id`):
- Player characters: `rpgm/element/character/pc`
- Non-player characters: `rpgm/element/character/npc`
- Factions: `rpgm/element/faction`
- Locations: `rpgm/element/location`
- Event: `rpgm/element/event`
- Clue: `rpgm/element/clue`

Every outline or element (excluding the campaign) MUST contain the tag `rpgm/campaign/{campaignId}`

I think there is a bit of added complexity, but I am considering placing the frontmeta templates in the code, so to be able to create it dynamically.

If you pull the `dev` branch you can look into it. Any feedback is more than welcome.
An example working with the updated version can be found [here](https://www.dropbox.com/sh/c7lbe6ms9qlkwue/AAAPJr-GabiT4QPsZooZeMA9a?dl=0).

**Comment by carlonicora** [comment]
There is a bug in my logic in regards to the navigation... The navigators won't work at the moment and I will be looking for a solution.

**TL;DR**: with the current tag setup, I can't uniquely get the adventure from a scene.

**Comment by carlonicora** [comment]
The outline tags require the creation of an id and the specification of the id of the parent.

- Campaigns: `rpgm/outline/campaign/{campaignId}
- Adventures: rpgm/outline/adventure/{adventureId}
- Sessions: rpgm/outline/session/{adventureId}/{sessionId}
- Scenes: rpgm/outline/scene/{adventureId}/{sessionId}/{sceneId}

The elements require a specific tag to identify their types (but don't require an id):

- Player characters: rpgm/element/character/pc
- Non-player characters: rpgm/element/character/npc
- Factions: rpgm/element/faction
- Locations: rpgm/element/location
- Event: rpgm/element/event
- Clue: rpgm/element/clue

Every outline or element (excluding the campaign) MUST contain the tag rpgm/campaign/{campaignId}
